tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -296,11 +297,98 @@ def test_slurm_protected_mode(
     )
     pending_job_id = _test_active_job_running(scheduler_commands, remote_command_executor, clustermgtd_conf_path)
     _test_protected_mode(scheduler_commands, remote_command_executor, cluster)
+
+    cw_client = boto3.client("cloudwatch", region_name=region)
+    cluster_name = cluster.cfn_name
+
+    _test_protected_mode_metric(cw_client, cluster_name)
+    _test_protected_mode_alarm(cw_client, cluster_name)
+
     test_cluster_health_metric(["NoCorrespondingInstanceErrors", "OnNodeStartRunErrors"], cluster.cfn_name, region)
     _test_job_run_in_working_queue(scheduler_commands)
     _test_recover_from_protected_mode(pending_job_id, pcluster_config_reader, bucket_name, cluster, scheduler_commands)
 
 
+@retry(stop_max_attempt_number=8, wait_fixed=minutes(2))
+def _test_protected_mode_metric(cw_client, cluster_name):
+    # query for the past 20 minutes
+    start_timestamp, end_timestamp = _get_start_end_timestamp(minutes=20)
+
+    protected_mode_values = _get_metric_data(cluster_name, cw_client, start_timestamp, end_timestamp)
+
+    assert_that(protected_mode_values).is_not_empty()
+
+
+def _test_protected_mode_alarm(cw_client, cluster_name):
+    protected_mode_alarm_name = f"{cluster_name}_ProtectedModeAlarm_HeadNode"
+
+    alarm_response = cw_client.describe_alarms(AlarmNames=[protected_mode_alarm_name])
+
+    protected_mode_alarm = _get_alarm_records(alarm_response, protected_mode_alarm_name)
+    _verify_alarms(protected_mode_alarm, "ClusterInProtectedMode", cluster_name)
+
+
+def _get_start_end_timestamp(minutes):
```

Comment:
same here, since this function is also used in test_monitoring.py. What about move it to utils.py and re-use it from there?

Reply:
Sure!